src/UIComponents/SelectDataset.jsx

```diff
@@ -64,11 +66,19 @@ class SelectDataset extends Component {
 
   handleUploadSelect = event => {
     this.props.resetState();
-    this.props.setSelectedCSV(event.target.files[0]);
+    const selectedCSV = event.target.files[0]
+    this.props.setSelectedCSV(selectedCSV);
     this.setState({
       download: false
     });
-    parseCSV(event.target.files[0], false, true);
+    if (selectedCSV.name.includes(".csv") ||
+    selectedCSV.type === "text/csv") {
```

Comment:
Unfortunately, it sounds like this change might actually cause unnecessary failures on some systems: https://github.com/mholt/PapaParse/issues/18